rest-api-spec/src/main/resources/rest-api-spec/test/cluster.allocation_explain/10_basic.yaml

```diff
@@ -0,0 +1,75 @@
+---
+"cluster shard allocation explanation test":
+  - do:
+      # there aren't any unassigned shards to explain
+      catch: request
+      cluster.allocation_explain: {}
+
+  - do:
+      indices.create:
+        index: test
+
+  - do:
```

Comment:
Maybe add a comment about how this relies on there only being a single node in the cluster. That is the case for the tests now but if someone tries to run them against a multi node cluster and doesn't understand what the API does they are going to be scratching their heads for a while.